### PR TITLE
hc(ref): Ensure OrganizationMember write paths have outboxes.

### PR DESCRIPTION
**Pull request by corps**
Based on https://github.com/getsentry/sentry/pull/49386 with minor adjustments.  Thanks @dashed !

Gets all organization member write paths completely covered, validated with the conftest blacklist.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_member/details.py [hunk 1]
```diff
@@ -296,8 +296,8 @@ def _change_org_role(member: OrganizationMember, role: str) -> None:
             ).update(role=None)
 
             member.update(role=role)
```
Comment (leeandher) [inline src/sentry/api/endpoints/organization_member/details.py:298]: Should this line be removed in favour of the new ones?

File: tests/conftest.py [hunk 2]
```diff
@@ -191,6 +192,12 @@ def protect_hybrid_cloud_deletions(request):
             seen_models.add(fk_model)
             restrict_role(role="postgres_unprivileged", model=fk_model, revocation_type="DELETE")
 
+    # Protect organization members from being updated without also invoking the correct outbox logic.
+    # If you hit test failures as  aresult of lacking these privileges, first ensure that you create the correct
```
Comment (leeandher) [inline tests/conftest.py:196]:
```suggestion
    # If you hit test failures as a result of lacking these privileges, first ensure that you create the correct
```

small typo, also the fact that you can restrict this is very cool

File: src/sentry/testutils/factories.py [hunk 3]
```diff
@@ -326,11 +328,11 @@ def create_team_membership(team, member=None, user=None, role=None):
...
-            if created:
-                member.outbox_for_create().drain_shard(max_updates_to_drain=10)
-                organizationmember_mapping_service.create_with_organization_member(
-                    org_member=member
-                )
+            # if created:
+            #     member.outbox_for_create().drain_shard(max_updates_to_drain=10)
+            #     organizationmember_mapping_service.create_with_organization_member(
+            #         org_member=member
+            #     )
```
Comment (markstory) [inline src/sentry/testutils/factories.py:335]: Should this be deleted instead of commented out?

Comment (corps) [reply src/sentry/testutils/factories.py:335]: Fixing these yes

**Review by markstory — APPROVED** [review]

**Pull request merged**